File: OpenDataFramework/EntityDataService.Utility/ExpandMethod.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace EntityDataService.Utility
{
    public static class ExpandMethod
    {
        public static string GetStringValue(this Dictionary<string, object> dict, string propertyName)
        {
            if (dict.ContainsKey(propertyName) && dict[propertyName] != null)
            {
                return dict[propertyName].ToString();
            }
            return string.Empty;
        }

        public static T GetValue<T>(this Dictionary<string, object> dict, string propertyName)
        {
            if (dict.ContainsKey(propertyName) && dict[propertyName] != null)
            {
                if (dict[propertyName] is T)
                {
                    return (T)dict[propertyName];
                }
                else
                {
                    object value = Common.ChangeType(dict[propertyName], typeof(T));
                    if (value != null)
                    {
                        return (T)value;
                    }
                }
            }
            return default(T);
        }

        public static string GetValue(this Dictionary<string, string> dict, string propertyName)
        {
            if (dict.ContainsKey(propertyName) && !string.IsNullOrEmpty(dict[propertyName]))
            {
                return dict[propertyName].Trim();
            }
            return string.Empty;
        }

        public static T GetValue<T>(this Dictionary<string, string> dict, string propertyName)
        {
            if (dict.ContainsKey(propertyName) && !string.IsNullOrEmpty(dict[propertyName]))
            {
                object value = Common.ChangeType(dict[propertyName], typeof(T));
                if (value != null)
                {
                    return (T)value;
                }
            }
            return default(T);
        }
    }
}

File: OpenDataApi/Api/src/main/OpenDataAccess/Utility/ExpandMethod.java
package OpenDataAccess.Utility;

import java.util.Map;

public class ExpandMethod {
    public static String GetStringValue(Map<String, Object> dict, String propertyName) {
        if (dict.containsKey(propertyName) && dict.get(propertyName) != null) {
            return dict.get(propertyName).toString();
        }
        return "";
    }

    public static Object GetValue(Map<String, Object> dict, String propertyName) {
        if (dict.containsKey(propertyName) && dict.get(propertyName) != null) {
            return dict.get(propertyName);
        }
        return null;
    }

    public static <T> T GetValueToType(Class<T> cls, Map<String, Object> dict, String propertyName) throws Exception {
        if (dict.containsKey(propertyName) && dict.get(propertyName) != null) {
            Object value = dict.get(propertyName);
            if (dict.get(propertyName).getClass().equals(cls.getClass())) return (T) value;
            else {
                value = Common.ChangeType(cls, value);
                if (value != null) return (T) value;
            }
        }
        return (T) Common.GetTypeDefaultValue(cls);
    }
}